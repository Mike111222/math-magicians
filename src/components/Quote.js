import React, { useEffect, useState } from 'react';

function Quote() {
  const [quoteData, setQuoteData] = useState(null);
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          headers: {
            'X-Api-Key': 'KgdctuCDA+3eWnazYFlg0A==APQxUuKIIG5IDuJp',
          },
        });

        if (!response.ok) {
          throw new Error('Failed to fetch quote');
        }

        const data = await response.json();
        const { quote, author } = data[0];

        setQuoteData({ quote, author });
        setIsLoading(false);
      } catch (error) {
        setError(error.message);
        setIsLoading(false);
      }
    };

    fetchQuote();
  }, []);

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error:
        {error}
      </div>
    );
  }

  return (
    <div>
      <h2>Quote of the Day</h2>
      <p>{quoteData.quote}</p>
      <p>
        -
        {quoteData.author}
      </p>
    </div>
  );
}

export default Quote;
