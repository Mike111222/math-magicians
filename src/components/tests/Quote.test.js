import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../Quote';

it('render correctly and mathes the snapshot', () => {
  const { asFragment } = render(<Quote />);
  expect(asFragment()).toMatchSnapshot();
});
