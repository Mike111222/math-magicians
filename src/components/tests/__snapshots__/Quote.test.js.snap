// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly and mathes the snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
